### ISSUE TITLE
Be able to drag pages to where you want them in the side bar

### ISSUE THREAD

**Issue by K4LCIFER** [xournalpp/xournalpp#3724]
**Is your feature request related to a problem? Please describe.**
It's somewhat of a pain to be able to organize pages in the side bar. Currently you have to right click on the page that you want to move, and select "move up" or "move down" from the right click menu which will then move the page up or down a single position.

**Describe the solution you'd like**
I would like to be able to just click and drag the page up or down in the side panel to where I want.

**Comment by rolandlo** [comment]
The would improve the UX, I agree.

**Comment by forgottosave** [comment]
I think this would be very useful, too, and will take a look at this in the upcoming days.

**Comment by piegamesde** [comment]
Relatedly, please provide a way to organize pages that does not involve the side bar (some entry in the menus would be sufficient). I have the side bar disabled for space reasons, and did not even realize this feature existed. It is not discoverable.

**Comment by forgottosave** [comment]
Yes, I also agree that in theory every important action should be reachable in the menu. Do you think a simple entry like `Journal > Move current page up / down` would be useful?

**Comment by piegamesde** [comment]
Yes, absolutely. On the one hand, I'd probably look for it under "Journal" first, because that's where the "New page" stuff is. On the other hand I should mention that I've seen the "Edit -> Arrange" menu, and then had hoped this was it just to find out it's about the drawing order.

As a bonus, a "move page to page number" entry would be nice to have. It may be of rare you so does not need a button in the side bar, but in the cases where you have to move a page to the end or something clicking "move down" twenty times is not that great :D

**Comment by forgottosave** [comment]
If the maintainers approve of this, I could quickly implement these menu entries
